modepy/quadrature/__init__.py

```diff
@@ -82,3 +102,79 @@ def __init__(self, quad, left, right):
         Quadrature.__init__(self,
                 left + (quad.nodes+1) / 2 * length,
                 quad.weights * half_length)
+
+
+class TensorProductQuadrature(Quadrature):
```

Comment:
We've got another case of "1D is special" here, where (for example) a `TensorProductQuadrature` with one entry in `quads` works differently from the 1D quadratures it takes as input.

Reply:
The only difference seems to be that `nodes` is of shape `(1, nnodes)` instead of just `nnodes`, right?

Any preference for a fix? I can add a `squeeze` in there or implement `__new__` and just return the 1D quadrature back?

Reply:
I would kind of like to push in the opposite direction, of everything being dimension-explicit, even the 1D things. Of course that'll make 1D things a tiny bit more awkward, but people can squeeze pretty easily. How about introducing a flag to the 1D quadrature getters (like `force_dim_axis=False`) and deprecating the `force_dim_axis=False` case?

Reply:
Which getters do you mean? `JacobiGaussQuadrature` and friends?

Reply:
Yeah. Everyone that produces a 1D quadrature.

Reply:
877018e

Hopefully I didn't forget anyone.

Reply:
LGTM